lib/connectors/memory.js

```diff
@@ -54,10 +55,38 @@ Memory.prototype.getTypes = function() {
 };
 
 Memory.prototype.connect = function(callback) {
+  var self = this;
+  if (this.connected) {
+    // It's already connected or being connected
+    // Do NOT run loadFromFile again as it will override the cache
+    if (callback) process.nextTick(callback);
+    return;
+  }
+  if (callback) {
+    // Queue the callback
+    this.pendingConnectCallbacks = this.pendingConnectCallbacks || [];
+    this.pendingConnectCallbacks.push(callback);
+  }
+  // The connect is already in progress
+  if (this.connecting) return;
+  // Set connecting flag to be true
+  this.connecting = true;
   if (this.isTransaction) {
     this.onTransactionExec = callback;
   } else {
-    this.loadFromFile(callback);
+    this.loadFromFile(function(err) {
+      if (!err) self.connected = true;
+      self.connecting = false;
+      // Invoke all pending callbacks
+      async.each(self.pendingConnectCallbacks, function(cb, done) {
+        process.nextTick(function() {
+          cb(err);
+          done();
+        });
```

Comment:
Isn't `async.each` already inserting `process.nextTick` calls? Would it make sense to do the following?

```js
function(cb, done) {
  process.nextTick(cb, err);
  done();
}
```